doc/tuto.md

```diff
@@ -69,8 +69,8 @@ let test file =
   (* We can loop over the parsed statements to generated the typed statements *)
   let final_state, rev_typed_stmts =
     List.fold_left (fun (state, acc) parsed_stmt ->
-      let state, typed_stmt = Typer.check state parsed_stmt in
-      (state, typed_stmt :: acc)
+      let state, typed_stmts = Typer.check state parsed_stmt in
+      (state, List.rev_append typed_stmts acc)
```

Comment:
Is this `rev_append` yielding the correct final order? In what order should we process the `typed_stmts` that come out of `Typer.check`?

Reply:
It should yield the correct order **ultimately**, indeed note that this fold produces the list of statements **in reverse order**, which is underscored by the fact that the result is bound to a variable named `rev_typed_stmts`.

The statements returned by `check` should be treated in the natural order of the list (one should be able to print the statements in the order provided, and get a problem equivalent to the one from the input).